### PR TITLE
[WIP] Fix default registry issue

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR addresses the issue of default registry.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: aea/cli/utils/click_utils.py [hunk 1]
```diff
@@ -208,32 +208,57 @@ def convert(self, value: str, param: Any, ctx: click.Context) -> str:  # type: i
...
         f = option(
-            "--ipfs",
+            "--remote",
             "registry",
-            flag_value=REGISTRY_IPFS,
-            help="For pushing items to ipfs node.",
-            default=(REGISTRY_IPFS == default_registry) and mark_default,
+            flag_value=REGISTRY_REMOTE,
+            help="To use a remote registry.",
+            default=(REGISTRY_REMOTE == default_registry) and mark_default,
```
Comment (angrybayblade) [inline aea/cli/utils/click_utils.py:230]: Now we have only 2 main types of registries, local and remote. Using local flag will use the local packages folder. Remote have 2 subtypes ipfs and http one of which can be set as default in global config when initialising aea.

File: aea/cli/utils/click_utils.py [hunk 2]
```diff
@@ -208,32 +208,57 @@ def convert(self, value: str, param: Any, ctx: click.Context) -> str:  # type: i
...
+            flag_value=REMOTE_IPFS,
+            help="To use a local registry.",
+            default=(REMOTE_IPFS == default_registry) and mark_default,
+        )(f)
+        f = option(
+            "--http",
+            "remote_registry",
+            flag_value=REMOTE_HTTP,
+            help="To use a remote registry.",
+            default=(REMOTE_HTTP == default_registry) and mark_default,
```
Comment (angrybayblade) [inline aea/cli/utils/click_utils.py:261]: To support http and ipfs registry subtypes

File: aea/cli/add.py [hunk 3]
```diff
@@ -205,6 +201,23 @@ def _add_item_deps(
...
+    if get_default_remote_registry() == REMOTE_IPFS:
+        try:
+            return fetch_ipfs(item_type, item_public_id, dest_path)
+        except HashNotProvided:
+            click.echo(f"Hash was not provided for: {item_public_id}")
+            click.echo("Will try with http repository.")
+
+    return fetch_package(
+        item_type, public_id=item_public_id, cwd=ctx.cwd, dest=dest_path,
+    )
```
Comment (angrybayblade) [inline aea/cli/add.py:218]: If fetching from IPFS fails should we try with http or let the user specify http while trying again?

File: aea/cli/init.py [hunk 4]
```diff
@@ -28,53 +30,84 @@
...
+)
 
 
 @click.command()
 @click.option("--author", type=str, required=False)
 @click.option("--reset", is_flag=True, help="To reset the initialization.")
 @click.option("--no-subscribe", is_flag=True, help="For developers subscription.")
 @registry_flag(mark_default=False)
+@remote_registry_flag(mark_default=False)
+@click.option("--ipfs-node", type=str, default=None, help="Multiaddr for IPFS node.")
```
Comment (angrybayblade) [inline aea/cli/init.py:56]: If a user wants to set multiaddr for IPFS node

File: aea/cli/scaffold.py [hunk 5]
```diff
@@ -225,6 +220,16 @@ def scaffold_item(ctx: Context, item_type: str, item_name: str) -> None:
...
+        package_hash = IPFSHashOnly().hash_directory(str(dest))
+        new_public_id_with_hash = PublicId(
+            author_name, item_name, DEFAULT_VERSION, package_hash
+        )
+        if not to_local_registry:
+            logger.debug(f"Registering the {item_type} into {DEFAULT_AEA_CONFIG_FILE}")
+            existing_ids.add(new_public_id_with_hash)
+            with open_file(os.path.join(ctx.cwd, DEFAULT_AEA_CONFIG_FILE), "w") as fp:
+                ctx.agent_loader.dump(ctx.agent_config, fp)
+
```
Comment (angrybayblade) [inline aea/cli/scaffold.py:232]: This change was supposed to be on the base PR directly but I forgot to commit before checking out to this branch 😅. While scaffolding we need to use the new public id format to register the newly scaffolded component in `aea-config.yaml`, this allows us to do that.

**Review by DavidMinarsch — APPROVED** [review]

File: aea/cli/add.py (hunk 3 above)
Comment (DavidMinarsch) [reply aea/cli/add.py:218]: Sure let's try the other remote, why not

**Pull request merged**